core/services/relay/evm/mercury/v1/reportcodec/report_codec.go

```diff
@@ -68,44 +69,28 @@ func NewReportCodec(feedID [32]byte, lggr logger.Logger) *ReportCodec {
 	return &ReportCodec{lggr, feedID}
 }
 
-func (r *ReportCodec) BuildReport(paos []reportcodec.ParsedAttributedObservation, f int, validFromBlockNum int64) (ocrtypes.Report, error) {
-	if len(paos) == 0 {
-		return nil, errors.Errorf("cannot build report from empty attributed observations")
+func (r *ReportCodec) BuildReport(rf reportcodec.ReportFields) (ocrtypes.Report, error) {
+	var merr error
+	if rf.BenchmarkPrice == nil {
+		merr = errors.Join(merr, errors.New("benchmarkPrice may not be nil"))
 	}
-
-	mPaos := reportcodec.Convert(paos)
-
-	timestamp := relaymercury.GetConsensusTimestamp(mPaos)
-	benchmarkPrice, err := relaymercury.GetConsensusBenchmarkPrice(mPaos, f)
-	if err != nil {
-		return nil, errors.Wrap(err, "GetConsensusBenchmarkPrice failed")
+	if rf.Bid == nil {
+		merr = errors.Join(merr, errors.New("bid may not be nil"))
 	}
-	bid, err := relaymercury.GetConsensusBid(mPaos, f)
-	if err != nil {
-		return nil, errors.Wrap(err, "GetConsensusBid failed")
+	if rf.Ask == nil {
+		merr = errors.Join(merr, errors.New("ask may not be nil"))
 	}
-	ask, err := relaymercury.GetConsensusAsk(mPaos, f)
-	if err != nil {
-		return nil, errors.Wrap(err, "GetConsensusAsk failed")
+	if len(rf.CurrentBlockHash) != 32 {
```

Comment:
This check is always going to pass as CurrentBlockHash is a fixed size 32 byte array

Reply:
Not if we are on a non-evm chain. It's a sanity check